### DISCUSSION TITLE
Saving data on home assistant

### DISCUSSION THREAD

**Discussion by komax74** [jomjol/AI-on-the-edge-device#1282] [General]
Good morning everyone, congratulations really great project, I have been using for a month sending mqtt data to home assistant all ok, then I had a problem with the power supply and for a few days esp32 went offline and I took the opportunity to place it better. At the end of the new setting procedure the system started sending data to home assistant again but I was shocked because lost all the data history.
Is it not possible to save them so that everything that is sent remains stored?

**Comment by jomjol** [comment]
That will be done in the next release, where a data log will be written in a text file. With this you should be able to recover the data manually later on.

**Comment by caco3** [comment]
We should add a menu point for the folder => https://github.com/jomjol/AI-on-the-edge-device/issues/1284

**Comment by nliaudat** [comment]
For Hassio, the is a new [buildin integration for water usage](https://www.home-assistant.io/blog/2022/11/02/release-202211/#getting-insights-into-water-usage). I mean that the records cannot be loosed if you use that.
